app/controllers/groups_controller.rb

```diff
@@ -23,8 +23,11 @@ def new
   end
 
   def show
-    @group = Group.find(params[:id])
-    @admin = @group.admin? current_user
+    @group = Group.find params[:id]
+    @membership = Membership.new
+    @current_user_is_admin = @group.admin? current_user
+    @current_user_applied_to_group = @group.pending_application? current_user
+    @current_user = current_user
```

Comment:
I think moving this into a presenter object might serve us better. I always use the rule that controllers can only pass one instance variable to the view, and the view can only talk to that object.